src/hamiltonian.jl

```diff
@@ -7,7 +7,7 @@ end
 
 struct PhysicsTag end
 
-function HamiltonianProblem{T}(H,p0,q0,tspan,p=nothing;kwargs...) where T
+function HamiltonianProblem{T}(H,q0,p0,tspan,p=nothing;kwargs...) where T
```

Comment:
I think the `p,q` order wast intentionally set like this in order to be consistent with how the rest of the problems are defined. See https://sciml.ai/2018/01/24/Parameters.html for more details.

Reply:
Why is line 3 the opposite then?

Reply:
I think it was missed during the update. You can see the rest of the changes in [this](https://github.com/SciML/DiffEqPhysics.jl/commit/65c93463a818e3538851770ea69cf1849b0f458f) commit by @ChrisRackauckas.